rpcs3/rpcs3qt/vfs_dialog_usb_tab.cpp

```diff
@@ -117,23 +117,25 @@ void vfs_dialog_usb_tab::show_usb_input_dialog(int index)
 	}
 
 	const QString device_name = get_device_name(index);
-	const cfg::device_info default_info = get_device_info(device_name, m_cfg_node->get_default());
-	cfg::device_info info{};
+	const auto default_info = std::make_shared<cfg::device_info>(get_device_info(device_name, m_cfg_node->get_default()));
+	auto info = std::make_shared<cfg::device_info>();
 
-	info.path   = m_usb_table->item(index, usb_column::usb_path)->text().toStdString();
-	info.vid    = m_usb_table->item(index, usb_column::usb_vid)->text().toStdString();
-	info.pid    = m_usb_table->item(index, usb_column::usb_pid)->text().toStdString();
-	info.serial = m_usb_table->item(index, usb_column::usb_serial)->text().toStdString();
+	info->path   = m_usb_table->item(index, usb_column::usb_path)->text().toStdString();
+	info->vid    = m_usb_table->item(index, usb_column::usb_vid)->text().toStdString();
+	info->pid    = m_usb_table->item(index, usb_column::usb_pid)->text().toStdString();
+	info->serial = m_usb_table->item(index, usb_column::usb_serial)->text().toStdString();
 
-	vfs_dialog_usb_input* input_dialog = new vfs_dialog_usb_input(device_name, default_info, &info, m_gui_settings, this);
-	if (input_dialog->exec() == QDialog::Accepted)
+	vfs_dialog_usb_input* input_dialog = new vfs_dialog_usb_input(device_name, *default_info, info.get(), m_gui_settings, this);
```

Comment:
this is horrible.
you're passing a raw pointer here which is captured inside the dialog.
Either make the constuctor arguments shared or use const refs.

Reply:
Their ownership is captured in the lambda below.

Reply:
that's still horrible software design.
the code should be consistent in its parts, and if you just look at the tab code it makes no sense at all and will cause some regression in the future.

Reply:
Parent events are still executing even from QDialog::exec(), if som code does not currently doesn't abide it then it's a bug.